wpilibj/src/main/java/edu/wpi/first/wpilibj/experimental/controller/PIDControllerRunner.java

```diff
@@ -0,0 +1,114 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2018-2019 FIRST. All Rights Reserved.                        */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package edu.wpi.first.wpilibj.experimental.controller;
+
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.function.DoubleConsumer;
+import java.util.function.DoubleSupplier;
+
+import edu.wpi.first.wpilibj.Notifier;
+
+public class PIDControllerRunner {
```

Comment:
Should be final if the behavior cannot be modified by subclassing

```suggestion
public final class PIDControllerRunner {
```

Reply:
If the behavior can't be modified, then marking it as final is redundant. A better question is whether we want the behavior to be modifiable in the future.

Reply:
This class feels like a good candidate for marking as final.